src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -406,7 +406,7 @@ private static IncrementalStepRunReason AsStepState(EntryState inputState, Entry
                     (EntryState.Cached, EntryState.Cached) => IncrementalStepRunReason.Cached,
                     (EntryState.Removed, EntryState.Removed) => IncrementalStepRunReason.Removed,
                     (EntryState.Modified, EntryState.Removed) => IncrementalStepRunReason.Removed,
-                    (EntryState.Modified, EntryState.Added) => IncrementalStepRunReason.Modified,
+                    (EntryState.Modified, EntryState.Added) => IncrementalStepRunReason.New,
```

Comment:
Should we document the breaking change somewhere? Seems like this is the place, but not sure it's actually used: https://github.com/dotnet/roslyn/blob/main/docs/Breaking%20API%20Changes.md

Reply:
I added a new "Unreleased" section to that doc, as I'm not sure how your versioning works. If there's a concrete version number I should add, I can change that.